org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/proposals/InitializeFinalFieldProposal.java

```diff
@@ -34,11 +36,16 @@ public InitializeFinalFieldProposal(IProblemLocation problem, ICompilationUnit c
 	}
 
 	public InitializeFinalFieldProposal(IProblemLocation problem, ICompilationUnit cu, ASTNode astNode, int relevance, int updateType) {
-		super(Messages.format(CorrectionMessages.InitializeFieldInConstructorCorrectionProposal_description, problem.getProblemArguments()[0]), cu, null, relevance,
+		super(Messages.format(
+				updateType == InitializeFinalFieldProposalCore.UPDATE_CONSTRUCTOR_NEW_PARAMETER
+						? CorrectionMessages.InitializeFieldWithConstructorParameterCorrectionProposal_description
+						: CorrectionMessages.InitializeFieldAtDeclarationCorrectionProposal_description,
```

Comment:
This message should be CorrectionMessages.InitializeFieldInConstructorCorrectionProposal_description

Reply:
Indeed. Fixed this as well.

Reply:
> Actually, it is the 2nd message that is in error. The first should be ..WithConstructor.... as you have above and the 2nd should be ..InConstructor. There is no case here where ..AtDeclaration will be used - it is done in the separate constructor above.

Yes, I just noticed it myself and the fix is already provided. Sorry for this inconsistency.